pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorSubscriptionTest.java

```diff
@@ -230,6 +233,94 @@ public void testReplicationSnapshotStopWhenNoTraffic() throws Exception {
+        stats = admin1.topics().getStats(topicName);
+        assertFalse(stats.subscriptions.get(subName).isReplicated);
+        stats = admin2.topics().getStats(topicName);
+        assertTrue(stats.subscriptions.get(subName).isReplicated);
+
+        // Disable replicated subscription in r2
+        admin2.topics().setReplicatedSubscriptionStatus(topicName, subName, false);
+        stats = admin2.topics().getStats(topicName);
+        assertFalse(stats.subscriptions.get(subName).isReplicated);
+
+        // Unload topic in r1
+        admin1.topics().unload(topicName);
+        stats = admin1.topics().getStats(topicName);
+        assertFalse(stats.subscriptions.get(subName).isReplicated);
+
+        // Enable replicated subscription in r1
+        admin1.topics().setReplicatedSubscriptionStatus(topicName, subName, true);
+        stats = admin1.topics().getStats(topicName);
+        assertTrue(stats.subscriptions.get(subName).isReplicated);
+        stats = admin2.topics().getStats(topicName);
+        assertFalse(stats.subscriptions.get(subName).isReplicated);
+    }
+
+    @Test(timeOut = 30000)
+    public void testReplicatedSubscriptionRestApi2() throws Exception {
+        final String namespace = BrokerTestUtil.newUniqueName("pulsar/replicatedsubscription");
+        final String topicName = "persistent://" + namespace + "/topic-rest-api2";
+        final String subName = "sub";
+
+        admin1.namespaces().createNamespace(namespace);
+        admin1.namespaces().setNamespaceReplicationClusters(namespace, Sets.newHashSet("r1"));
+        admin1.topics().createPartitionedTopic(topicName, 2);
+
+        @Cleanup
+        final PulsarClient client1 = PulsarClient.builder().serviceUrl(url1.toString())
+                .statsInterval(0, TimeUnit.SECONDS).build();
+
+        // Create subscription in r1
+        createReplicatedSubscription(client1, topicName, subName, true);
+
+        PartitionedTopicStats partitionedStats = admin1.topics().getPartitionedStats(topicName, true);
+        for (TopicStats stats : partitionedStats.partitions.values()) {
+            assertTrue(stats.subscriptions.get(subName).isReplicated);
+        }
+
+        // Disable replicated subscription in r1
+        admin1.topics().setReplicatedSubscriptionStatus(topicName, subName, false);
+        partitionedStats = admin1.topics().getPartitionedStats(topicName, true);
+        for (TopicStats stats : partitionedStats.partitions.values()) {
+            assertFalse(stats.subscriptions.get(subName).isReplicated);
+        }
+
+        // Enable replicated subscription in r1
+        admin1.topics().setReplicatedSubscriptionStatus(topicName, subName, true);
+        partitionedStats = admin1.topics().getPartitionedStats(topicName, true);
+        for (TopicStats stats : partitionedStats.partitions.values()) {
+            assertTrue(stats.subscriptions.get(subName).isReplicated);
+        }
```

Comment:
We are not testing that actually the replication is starting to work or it is stopping to work.
We are only testing on the value of the internal status.

Can we some end-to-end test ?
otherwise it would be possible that it looks like replication is enabled but actually it is not working

Reply:
Added some e2e tests.